File: core/src/main/java/com/zeyad/rxredux/core/redux/BaseView.java
package com.zeyad.rxredux.core.redux;

import android.content.Intent;
import android.os.Bundle;
import android.os.Parcelable;

/**
 * @author ZIaDo on 1/19/18.
 */
public class BaseView {

    public static final String UI_MODEL = "viewState";

    private BaseView() {
    }

    public static <S extends Parcelable> S getViewStateFrom(Bundle savedInstanceState, Intent intent) {
        S viewState = null;
        if (savedInstanceState != null && savedInstanceState.containsKey(UI_MODEL)) {
            viewState = savedInstanceState.getParcelable(UI_MODEL);
        } else if (intent != null && intent.hasCategory(UI_MODEL)) {
            viewState = intent.getParcelableExtra(UI_MODEL);
        }
        return viewState;
    }

    public static <S extends Parcelable> S getViewStateFrom(Bundle savedInstanceState, Bundle arguments) {
        S viewState = null;
        if (savedInstanceState != null && savedInstanceState.containsKey(UI_MODEL)) {
            viewState = savedInstanceState.getParcelable(UI_MODEL);
        } else if (arguments != null && arguments.containsKey(UI_MODEL)) {
            viewState = arguments.getParcelable(UI_MODEL);
        }
        return viewState;
    }
}

File: core/src/main/java/com/zeyad/rxredux/core/BaseView.kt
package com.zeyad.rxredux.core

import android.content.Intent
import android.os.Bundle
import android.os.Parcelable

/**
 * @author ZIaDo on 2/27/18.
 */
class BaseView {

    companion object {
        const val UI_MODEL = "viewState"

        fun <S : Parcelable> getViewStateFrom(savedInstanceState: Bundle?, intent: Intent?): S? =
                if (savedInstanceState != null && savedInstanceState.containsKey(UI_MODEL)) {
                    savedInstanceState.getParcelable(UI_MODEL)
                } else if (intent != null && intent.hasCategory(UI_MODEL)) {
                    intent.getParcelableExtra(UI_MODEL)
                } else null

        fun <S : Parcelable> getViewStateFrom(savedInstanceState: Bundle?, arguments: Bundle?): S? =
                if (savedInstanceState != null && savedInstanceState.containsKey(UI_MODEL)) {
                    savedInstanceState.getParcelable(UI_MODEL)
                } else if (arguments != null && arguments.containsKey(UI_MODEL)) {
                    arguments.getParcelable(UI_MODEL)
                } else null
    }
}

